// Message2Server
syntax = "proto3";
package protobuf;
import "MessageType.proto";

message IDMsg
{
    int64 player_id = 1;
    int64 team_id = 2;
}
message PlayerMsg
{
    int64 player_id = 1;
    ShipType ship_type = 2;
    int64 team_id = 3;
    int32 x=4;
    int32 y=5;
}

message MoveMsg
{
    int64 player_id = 1;
    double angle = 2;
    int64 time_in_milliseconds = 3;
    int64 team_id = 4;
}


message SendMsg
{
    int64 player_id = 1;
    int64 to_player_id = 2;
    oneof message
    {
        string text_message = 3;
        bytes binary_message = 4;
    }
    int64 team_id = 5;
}

message AttackMsg // 相当于攻击
{
    int64 player_id = 1;
    double angle = 2;
    int64 team_id = 3;
}

message ConstructMsg //修建建筑
{
    int32 x = 1;
    int32 y = 2;
    int64 player_id = 3;
    ConstructionType construction_type = 4;
    int64 team_id = 5;
}

message TargetMsg  
{
    int32 x = 1;
    int32 y = 2;
    int64 player_id = 3;
    int64 team_id = 4;
}
message InstallMsg
{
    ModuleType module_type = 1;
    int64 player_id = 2;
    int64 team_id = 3;
}
message BuildShipMsg
{
    int32 x = 1;
    int32 y = 2;
    ShipType ship_type = 3;
    int64 team_id = 4;
}